WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactables/House.cs

```diff
@@ -1,30 +1,53 @@
 ﻿using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
+using System;
 
-public class House : Interactable
+[RequireComponent(typeof(Interactable))]
+public class House : MonoBehaviour
 {
     private SpriteRenderer myRenderer;
+    private Interactable interactable;
+
+    protected Color _interactableColor = new Color(1f, 1f, 1f, .25f);
+    protected Color _defaultColor = new Color(1f, 1f, 1f, 0f);
+    protected Color _interactionColor = new Color(.5f, .8f, .2f, .25f);
+
 
     private void Awake()
     {
         myRenderer = GetComponent<SpriteRenderer>();
         myRenderer.color = _defaultColor;
+        interactable = GetComponent<Interactable>();
+    }
+
+    private void OnEnable()
+    {
+        interactable.GainedFocus += OnGainedFocus;
```

Comment:
I want to punch C#

Reply:
You might want to punch unity. It could be unity's fault.

Reply:
No it's just the overridden += for event subscription. Way too cute